src/v/bytes/tests/iobuf_fragment_list_test.cc

```diff
@@ -0,0 +1,198 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "bytes/details/io_fragment.h"
+
+#include <gtest/gtest-matchers.h>
+#include <gtest/gtest.h>
+
+#include <list>
+#include <ranges>
+
+using namespace details;
+
+namespace {
+
+TEST(FragmentListTests, Back) {
```

Comment:
Might be worth considering fuzzing the various push/pop methods, given their implementations are pretty non-trivial

Reply:
We actually do this in an indirect way in the iobuf_fuzz tests, which I've run extensively